### PR TITLE
Update noticon from notification to bell

### LINKED ISSUE #36192: Top Bar Refinements: Update the notification icon
The notification icon we use in wp-admin feels dated. It uses an orange gradient. Since this does not ship with core, we should update it to use the icon from nav unification

Screenshot 2024-03-05 at 17 54 26

Project Thread - pfsHM7-oC-p2

### PR DESCRIPTION
**Pull request by eoigal**
Fixes https://github.com/Automattic/jetpack/issues/36192

This updates the notification icon used in the top bar when the Masterbar is not being rendered.

## Changes

The only change here is to the notification HTML markup, its slightly different. 
It used to be;

```
<a class="ab-item" href="https://wordpress.com/notifications">
	<span id="wpnt-notes-unread-count" class="wpnt-loading wpn-read">
		<span class="screen-reader-text">Notifications</span>
		<span class="noticon noticon-notification"></span>
	</span>
</a>
```

And now it will be the same as on Masterbar;

```
<a class="ab-item" href="https://wordpress.com/notifications">
	<span id="wpnt-notes-unread-count" class="wpnt-loading wpn-read"></span>
	<span class="noticon noticon-bell"></span>
	<span class="screen-reader-text">Notifications</span>
</a>
```

This PR adds a new static function  `get_notes_markup` to return the markup used.

The CSS changes are done to the following CSS file;

`https://s0.wp.com/wp-content/mu-plugins/notes/admin-bar-v2.css`

Those changes are done on WPCOM sandbox on D141499-code.

We need to add a top level class to the `#wp-admin-bar-notes` li element so that we can support styling the bell icon while also supporting the exisiting notification icon on older verisons of Jetpack.

## Before

https://github.com/Automattic/jetpack/assets/5560595/8fc08e00-4346-45f8-a4ca-23fb20c06e48

## After

https://github.com/Automattic/jetpack/assets/5560595/7a34521b-e8ab-4516-97ac-defba0636659

## Testing instructions:

* Apply CSS changes in D141499-code
* Add `define( 'JETPACK_NOTES__CACHE_BUSTER', time() );` to your 0-sandbox.php file
* Follow setup guidelines here - pdWQjU-77-p2
* Create JN site
* Complete Jetpack setup and select free plan
* Confirm notification looks/behaves the same as notification icon on simple site
* Create JN site but don't sync PR, confirm the older notification icon looks/behaves ok (pictures below)
Screenshot 2024-03-14 at 12 25 37
Screenshot 2024-03-14 at 12 25 23

## Does this pull request change what data or activity we track or use?

No

### CODE REVIEW

**Comment by roo2** [comment]
There is still quite a bit of CSS to remove to get the flat look. As well as the :hover, :focus states

https://github.com/Automattic/jetpack/assets/22446385/bb98ebc2-5a67-4154-b2a3-3eb66da83dbe

cleanshot-2024-03-05-at-10 58 18402x

I got it looking a bit better in the browser dev tools by taking the background image off `#wp-admin-bar-notes #wpnt-notes-unread-count.wpn-read` (I had to use `!important)

Screenshot 2024-03-11 at 9 40 03 am

**Comment by eoigal** [comment]
> There is still quite a bit of CSS to remove to get the flat look. As well as the :hover, :focus states

Wasn't ready for review!.. it is now though 😄

File: projects/plugins/jetpack/modules/notes.php [hunk 1]
```diff
@@ -124,7 +124,7 @@ public function styles_and_scripts() {
 		}
 
 		if ( ! $is_rtl ) {
-			wp_enqueue_style( 'wpcom-notes-admin-bar', $this->wpcom_static_url( '/wp-content/mu-plugins/notes/admin-bar-v2.css' ), array( 'admin-bar' ), JETPACK_NOTES__CACHE_BUSTER );
+			wp_enqueue_style( 'wpcom-notes-admin-bar', $this->wpcom_static_url( '/wp-content/mu-plugins/notes/admin-bar-v2.css' ), array( 'admin-bar' ), JETPACK_NOTES__CACHE_BUSTER . '-' . time() );
```
Comment (eoigal) [inline projects/plugins/jetpack/modules/notes.php]: This will be removed before merging - just left it here to bust cache when testing CSS changes.

**Review by Addison-Stavlo — COMMENTED** [review]
I mentioned this on the diff, but same concern here. D141499-code 

We want to do this in a way where older versions of jetpack without the new bell icon (the change from this PR) still look the same. The CSS in the diff seems to be changing styles that are still relevant to older JP versions where the bell is not present. 

While we have a new `.noticon-bell` class here, I think you are applying/changing styles at a higher level than that. We may need to add a new class at the top of the wrapper  element to signify this change, and make our CSS changes dependent on that selector (so that only this version and higher of JP gets those style changes).

File: projects/plugins/jetpack/modules/notes.php [hunk 2]
```diff
@@ -136,7 +136,7 @@ public function styles_and_scripts() {
 		$script_handles = array();
 		wp_register_script( 'wpcom-notes-common', $this->wpcom_static_url( '/wp-content/mu-plugins/notes/notes-common-lite.min.js' ), array(), JETPACK_NOTES__CACHE_BUSTER, true );
 		$script_handles[] = 'wpcom-notes-common';
-		wp_enqueue_script( 'wpcom-notes-admin-bar', $this->wpcom_static_url( '/wp-content/mu-plugins/notes/admin-bar-v2.js' ), array( 'wpcom-notes-common' ), JETPACK_NOTES__CACHE_BUSTER, true );
+		wp_enqueue_script( 'wpcom-notes-admin-bar', $this->wpcom_static_url( '/wp-content/mu-plugins/notes/admin-bar-v2.js' ), array( 'wpcom-notes-common' ), JETPACK_NOTES__CACHE_BUSTER . '-' . time() , true );
```
Comment (Addison-Stavlo) [inline projects/plugins/jetpack/modules/notes.php]: Is this also for testing and removed before merging?

**Review by Addison-Stavlo — COMMENTED** [review]
Tests well on my end!

File: projects/plugins/jetpack/modules/notes.php (hunk 2 above)
Comment (eoigal) [reply projects/plugins/jetpack/modules/notes.php]: Yep. I'll update patch now.. its not needed in testing anyway, you can define `JETPACK_NOTES__CACHE_BUSTER` in 0-sandbox.php instead.

File: projects/plugins/jetpack/modules/notes.php [hunk 3]
```diff
@@ -205,6 +202,17 @@ public function admin_bar_menu() {
...
 
+	/**
+	 * Returns the HTML markup for used by notification in top bar
+	 *
+	 * @return string
+	 */
+	private static function get_notes_markup() {
+		return '<span id="wpnt-notes-unread-count" class="wpnt-loading wpn-read"></span>
+<span class="noticon noticon-bell"></span>
+<span class="screen-reader-text">' . __( 'Notifications', 'jetpack' ) . '</span>';
```
Comment (jeherve) [inline projects/plugins/jetpack/modules/notes.php]: I would recommend leaving the extra escaping in there as it was:

```suggestion
<span class="screen-reader-text">' . esc_html__( 'Notifications', 'jetpack' ) . '</span>';
```

Comment (jeherve) [reply projects/plugins/jetpack/modules/notes.php]: Done in 2452d8a863e2907b93a78677e541e71bf8554266

**Review by jeherve — APPROVED** [review]
this tests well for me, both with `trunk` and with this branch. 🚢

**Pull request merged**